### PR TITLE
User consent passed through to SDC

### PR DESCRIPTION
**Pull request by charleycampbell**
## What does this change?
We are passing through the user consent from the cookies banner to the API which requests the banners
## Why?
There is a requirement to be able to target users via the RRCP depending on their user consent status.  This changes requites a couple of PRS from other repos:
[Support-dotcom PR](https://github.com/guardian/support-dotcom-components/pull/1100)
[Support-admin-console](https://github.com/guardian/support-admin-console/pull/571)
## Testing
This has been tested end to end on CODE with the the other changes on support dotcom components and support admin console. We have set up banner test where the targeting is set to either reject all, accept all or all (which doesn't take the the consent status into account).

### CODE REVIEW

File: dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ import { setAutomat } from '../../lib/setAutomat';
 import { useIsInView } from '../../lib/useIsInView';
 import { useOnce } from '../../lib/useOnce';
 import type { TagType } from '../../types/tag';
+import { hasRequiredConsents } from '../SignInGate/displayRule';
```
Comment (tomrf1) [inline dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx:34]: Let's not reuse code from the SignInGate for this.
Given how simple `hasRequiredConsents` is, I think it's ok to duplicate that code in this file

Comment (charleycampbell) [reply dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx]: Sorted! Should we add this type of thing into a shared file?

**Review by tomrf1 — APPROVED** [review]
looks good!

File: dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx]: Generally the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) is a good pattern to follow. Two instances do not quite meet threshold for abstracting away :)

**Pull request merged**